shm/Alcohol.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Alcohol.hpp"
+
+Cargo& Alcohol::operator+=(const size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Alcohol::operator-=(const size_t amount) {
+    if (amount > amount_) {
+        amount_ = 0;
+        return *this;
+    }
+    amount_ -= amount;
+
+    return *this;
+}
+
+bool Alcohol::operator==(const Cargo& cargo) const {
+    return getName() == cargo.getName() &&
+           getBasePrice() == cargo.getBasePrice() &&
+           getPrice() == cargo.getPrice();
```

Comment:
Do you need to calll `getName()` , getbasePrice and getPrice for 'this' object? You may use `name_`, `basePrice_` and so on, because I guess you have direct access to them in child class.

Then you can have `name_ == cargo.getName() ...`

Reply:
Yes, but class Cargo doesnt have `price_` member, it has to be calculated so I decided to put getters to make it more clear